app/src/main/java/app/revanced/integrations/shared/checks/Check.java

```diff
@@ -0,0 +1,123 @@
+package app.revanced.integrations.shared.checks;
+
+import android.annotation.SuppressLint;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.text.Html;
+import android.widget.Button;
+import app.revanced.integrations.shared.Logger;
+import app.revanced.integrations.shared.Utils;
+import app.revanced.integrations.youtube.settings.Settings;
+
+import java.util.function.Function;
+
+import static android.text.Html.FROM_HTML_MODE_COMPACT;
+import static app.revanced.integrations.shared.StringRef.str;
+
+abstract class Check {
+    private static final Uri GOOD_SOURCE = Uri.parse("https://revanced.app");
+    private static final Function<Context, AlertDialog.Builder> CHECK_FAILED_DIALOG_BUILDER =
+            (context) -> new AlertDialog.Builder(context)
+                    .setCancelable(false)
+                    .setIcon(android.R.drawable.ic_dialog_alert)
+                    .setTitle(str("revanced_check_environment_failed_title"))
+                    .setPositiveButton(
+                            str("revanced_check_environment_dialog_open_official_source_button"),
+                            (dialog, which) -> {
+                                final var intent = new Intent(Intent.ACTION_VIEW, GOOD_SOURCE);
+                                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                                context.startActivity(intent);
+                            }
+                    ).setNegativeButton(
+                            str("revanced_check_environment_dialog_ignore_button"),
+                            (dialog, which) -> {
+                                final int current = Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.get();
+                                Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.save(current + 1);
+
+                                dialog.dismiss();
+                            }
+                    );
+
+    private final String failedReasonStringKey;
+
+    /**
+     * @param failedReasonStringKey The string key of the reason why the check failed to be shown to the user.
+     */
+    Check(String failedReasonStringKey) {
+        this.failedReasonStringKey = failedReasonStringKey;
+    }
+
+    protected abstract boolean run();
+
+    static boolean shouldRun() {
+        return Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.get() < 2;
+    }
+
+    static void disableForever() {
+        Logger.printDebug(() -> "Environment checks disabled forever.");
+
+        Settings.CHECK_ENVIRONMENT_WARNING_ISSUED_COUNT.save(Integer.MAX_VALUE);
+    }
+
+    @SuppressLint("NewApi")
+    static void issueWarning(Context context, Check... failedChecks) {
```

Comment:
@LisoUseInAIKyrios The dialog is shown below other dialogs, is there any way to force it to show up on top or block other dialogs?